pkg/endpoint/bpf.go

```diff
@@ -1120,28 +1126,23 @@ func (e *Endpoint) addPolicyKey(keyToAdd policy.Key, entry policy.MapStateEntry,
 
 // ApplyPolicyMapChanges updates the Endpoint's PolicyMap with the changes
 // that have accumulated for the PolicyMap via various outside events (e.g.,
-// identities added / deleted).
-// 'proxyWaitGroup' may not be nil.
-func (e *Endpoint) ApplyPolicyMapChanges(proxyWaitGroup *completion.WaitGroup) error {
+// identities added / deleted) by triggering the sync-policy-map controller.
+// Returns a channel that is closed when the changes are applied. The channel
+// is closed even if errors occurred. The controller will retry the changes.
```

Comment:
does this mean the channel is closed before the policy is plumbed in case of errors? Or just that it is eventually closed, even in the face of errors?

Reply:
The old semantics it returned an error that was logged and then unblocked FQDN, so that's what I'm keeping here: Close the channel when an attempt has been made to reconcile the policy map changes. 

It's a separate topic whether that should change and we should keep waiting for retry to succeed in `notifyOnDNSMsg`.